tests/Fixer/FunctionNotation/PhpdocToReturnTypeFixerTest.php

```diff
@@ -0,0 +1,211 @@
+                '<?php /** @return callable */ function my_foo() {}',
+            ],
+            'self accessor' => [
+                '<?php
+                    class Foo {
+                        /** @return self */ function my_foo(): self {}
+                    }
+                ',
+                '<?php
+                    class Foo {
+                        /** @return self */ function my_foo() {}
+                    }
+                ',
+            ],
+            'report static as self' => [
+                '<?php
+                    class Foo {
+                        /** @return static */ function my_foo(): self {}
+                    }
+                ',
+                '<?php
+                    class Foo {
+                        /** @return static */ function my_foo() {}
+                    }
+                ',
+            ],
+            'skip resource special type' => [
+                '<?php /** @return resource */ function my_foo() {}',
+            ],
+            'skip mixed special type' => [
+                '<?php /** @return mixed */ function my_foo() {}',
+            ],
+            'null alone cannot be a return type' => [
+                '<?php /** @return null */ function my_foo() {}',
+            ],
+            'skip mixed types' => [
+                '<?php /** @return Foo|Bar */ function my_foo() {}',
+            ],
+            'nullable type' => [
+                '<?php /** @return null|Bar */ function my_foo(): ?Bar {}',
+                '<?php /** @return null|Bar */ function my_foo() {}',
+                70100,
+            ],
+            'nullable type reverse order' => [
+                '<?php /** @return Bar|null */ function my_foo(): ?Bar {}',
+                '<?php /** @return Bar|null */ function my_foo() {}',
+                70100,
+            ],
+            'nullable native type' => [
+                '<?php /** @return null|array */ function my_foo(): ?array {}',
+                '<?php /** @return null|array */ function my_foo() {}',
+                70100,
+            ],
+            'skip mixed nullable types' => [
+                '<?php /** @return null|Foo|Bar */ function my_foo() {}',
+            ],
+            'skip generics' => [
+                '<?php /** @return array<int, bool> */ function my_foo() {}',
+            ],
+            'array of types' => [
+                '<?php /** @return Foo[] */ function my_foo(): array {}',
+                '<?php /** @return Foo[] */ function my_foo() {}',
+            ],
+            'skip array of array of types' => [
+                '<?php /** @return Foo[][] */ function my_foo() {}',
+            ],
+            'nullable array of types' => [
+                '<?php /** @return null|Foo[] */ function my_foo(): ?array {}',
+                '<?php /** @return null|Foo[] */ function my_foo() {}',
+                70100,
+            ],
```

Comment:
I think it would be nice to have some complex cases, like
```php
class A
{
    /** @return Foo*/  # some comment
   final/**/static/**/function/**/A/**/(/**/)/**/{#comment 2
   }
}
```

Reply:
Thanks, added